extensions/amp-form/0.1/amp-form.js

```diff
@@ -769,12 +773,19 @@ export class AmpForm {
   }
 
   /**
-   * Transition the form to the submit success state.
+   * Transition the form to the submit-success or submit-error state depending on the response status.
    * @param {!JsonObject} response
    * @return {!Promise}
    * @private
    */
-  handleSsrTemplateSuccess_(response) {
+  handleSsrTemplateResponse_(response) {
+    if (
+      response.init &&
+      response.init.status &&
+      response.init.status > SUBMIT_ERROR_STATUS_THRESHOLD
+    ) {
+      return this.handleSubmitFailure_(response.init.status, response);
```

Comment:
What does the console user error look like in this case? "Form submission failed: 4xx"? 

Can we make this clearer?

Reply:
As for now, we are not passing anything useful from gmail side, but we are planning to do so. Maybe the status code is the best we can give here.

Reply:
In Dynamic Mail the original error gets filtered out in this response b/136171476
May be useful to define a standard response structure across email providers in order to propagate context.